Guidance to run throughput tool in docker shell (#6122)
Particular/docs.particular.net: nservicebus/throughput-tool/index.md

    Particular.EndpointThroughputCounter.exe [command] [options]
    ```

> ### Docker

Author: ```suggestion
### Containers
```

> If the system is running in Docker containers, the throughput tool can be run from a container that is based on the [Microsoft .NET SDK `mcr.microsoft.com/dotnet/sdk6.0` image](https://hub.docker.com/_/microsoft-dotnet-sdk/). Some systems will have such a container provisioned to access for system administration tasks.

Author: ```suggestion
If the system is running in containers, the throughput tool can be run from a container that is based on the [Microsoft .NET SDK `mcr.microsoft.com/dotnet/sdk6.0` image](https://hub.docker.com/_/microsoft-dotnet-sdk/). Some systems will have such a container provisioned to access for system administration tasks.
```

> 1. Open a remote shell using the following command:

Author: ```suggestion
1. Launch a container based on `mcr.microsoft.com/dotnet/sdk6.0` and open a remote bash to this container instance:
```

   ```shell

>    docker run -it --name=throughputtool mcr.microsoft.com/dotnet/sdk:6.0 bash

Author: ```suggestion
   // For example, use Docker and launch a bash shell
   docker run -it --name=throughputtool mcr.microsoft.com/dotnet/sdk:6.0 bash
```

   ```
1. Use the following command to install the throughput counter from MyGet:
    ```shell
    dotnet tool install -g Particular.EndpointThroughputCounter --add-source=https://www.myget.org/F/particular/api/v3/index.json
    ```
1. Run the tool by executing `throughput-counter`:
    ```shell
    throughput-counter [command] [options]
    ```
1. When the tool completes, find the JSON report file output by the tool:

>    ```shell
>    ls *.json
>    ```

Author: ```suggestion
    ```shell
    ls *.json
    ```
```

1. Transfer the file to your local system so it can be sent to Particular Software.
   ```shell

>    TODO

Author: ```suggestion
    // Write report to console to copy and paste this in a local text file in any editor of choice
    cat customer-name-YYYYMMDD-HHMMSS.json
```

    ```shell
    dotnet tool install -g Particular.EndpointThroughputCounter --add-source=https://www.myget.org/F/particular/api/v3/index.json
    ```
1. Run the tool by executing `throughput-counter`:
    ```shell
    throughput-counter [command] [options]
    ```
1. When the tool completes, find the JSON report file output by the tool:
    ```shell
    ls *.json
    ```
1. Transfer the file to your local system so it can be sent to Particular Software.
    ```shell
    # Example: Use cat to write the report to the console. This might not be ideal when dealing with large reports.
    cat ./customer-name-YYYYMMDD-HHMMSS.json

    # IMPORTANT: The following examples use 3rd party services. Review and accept their privacy policies regarding data privacy and removal concerns
    
    # Example: Upload the via https://transfer.sh, navigate to the returned url, download the report, and delete the file from the service.
    curl --upload-file ./customer-name-YYYYMMDD-HHMMSS.json https://transfer.sh

    # Example: Upload the via https://www.keep.sh/, navigate to the returned url, download the report, and delete the file from the service.

>     curl --upload-file ./customer-name-YYYYMMDD-HHMMSS.json https://www.keep.sh/

Reviewer: We can't do any of this. Because we require this file for licensing, anything that relies on copy/paste inside a terminal is a non-starter because it's too unreliable, as is anything that relies on a file transfer to an untrusted 3rd party as it's a data privacy concern.

Author: Why is that unreliable? Its json, I can SSH into my NAS and cat a huge text file and very easily copy & paste it from the shell to a text file. Even if it affects any white space that should not matter for utf8 text.

The privacy concern is there, that is why I mention the "IMPORTANT". Maybe we should just remove the ACTUAL examples and just define bullets of options?

>     ```shell
>     # Example: Use cat to write the report to the console. This might not be ideal when dealing with large reports.
>     cat ./customer-name-YYYYMMDD-HHMMSS.json
> 
>     # IMPORTANT: The following examples use 3rd party services. Review and accept their privacy policies regarding data privacy and removal concerns
>     
>     # Example: Upload the via https://transfer.sh, navigate to the returned url, download the report, and delete the file from the service.
>     curl --upload-file ./customer-name-YYYYMMDD-HHMMSS.json https://transfer.sh
> 
>     # Example: Upload the via https://www.keep.sh/, navigate to the returned url, download the report, and delete the file from the service.
>     curl --upload-file ./customer-name-YYYYMMDD-HHMMSS.json https://www.keep.sh/
>     ```

Author: ```suggestion
  - Suggestions to transfer the report json file:
    a. Mount a folder/volume to the container
    b. Use `cat` to write the report to the console and copy this to a file
    c. Use an untrusted 3rd party file transfer service like https://transfer.sh or https://www.keep.sh/
```
